src/__tests__/index.test.js

```diff
@@ -206,4 +206,40 @@ describe('denormalize', () => {
     });
     expect(() => denormalize('123', article, entities)).not.toThrow();
   });
+
+  it('denormalizes recursive dependencies', () => {
+    const user = new schema.Entity('users');
+    const report = new schema.Entity('reports');
+
+    user.define({
+      reports: [ report ]
+    });
+    report.define({
+      user: user
+    });
+
+    const entities = {
+      reports: {
+        1: {
+          title: 'Weekly report',
+          user: 1
+        },
+        2: {
```

Comment:
Why do you have all of this extra setup data (report 2, user 2). Can delete since it's not being used.